cmd/thv-operator/controllers/virtualmcpserver_discovery.go

```diff
@@ -0,0 +1,227 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	mcpv1alpha1 "github.com/stacklok/toolhive/cmd/thv-operator/api/v1alpha1"
+	ctrlutil "github.com/stacklok/toolhive/cmd/thv-operator/pkg/controllerutil"
+)
+
+const (
+	backendStatusReady       = "ready"
+	backendStatusDegraded    = "degraded"
+	backendStatusUnavailable = "unavailable"
+)
+
+// backendHealthResult represents the health status of all backends
+type backendHealthResult struct {
+	allHealthy       bool // All backends are healthy
+	someHealthy      bool // At least one backend is healthy
+	totalCount       int  // Total number of backends
+	unavailableCount int  // Number of unavailable backends
+}
+
+// discoverBackends discovers backend MCPServers from the MCPGroup and updates status
+func (r *VirtualMCPServerReconciler) discoverBackends(
```

Comment:
Is this needed? Discovery would happen inside the `vmcp` itself, so the reconcile doesn't necessarily need to know this.